src/Adapter/Category/CommandHandler/AbstractDeleteCategoryHandler.php

```diff
@@ -72,4 +76,140 @@ protected function handleProductsUpdate($parentCategoryId, CategoryDeleteMode $m
+
+        return $productsWithoutCategory ?? [];
+    }
+
+    /**
+     * @todo: move to repository sql part of the method - findProductsInDefaultCategories()?
+     *
+     * @param array<int, int[]> $deletedCategoryIdsByParent
+     *
+     * @return array
+     */
+    private function findProductsByDefaultCategories(array $deletedCategoryIdsByParent): array
+    {
+        $deletedCategoryIds = [];
+        foreach ($deletedCategoryIdsByParent as $deletedIds) {
+            $deletedCategoryIds = array_merge($deletedCategoryIds, array_map('intval', $deletedIds));
+        }
+
+        $affectedProducts = \Db::getInstance()->executeS('
+			SELECT p.`id_product`
+			FROM `' . _DB_PREFIX_ . 'product` p
+			' . Shop::addSqlAssociation('product', 'p') . '
+			WHERE p.id_category_default IN (' . implode(',', $deletedCategoryIds) . ')
+		');
+
+        return $affectedProducts ?? [];
+    }
+
+    /**
+     * @param int $categoryId
+     * @param array<int, int[]> $deletedCategoryIdsByParent
+     *
+     * @return int|null
+     */
+    private function findCategoryParentId(int $categoryId, array $deletedCategoryIdsByParent): ?int
+    {
+        $parentCategoryId = null;
+        foreach ($deletedCategoryIdsByParent as $parentId => $deletedIds) {
+            // find parent id for deleted category
+            if (in_array($categoryId, $deletedIds, true)) {
+                $parentCategoryId = $parentId;
+                break;
+            }
+        }
+
+        return $parentCategoryId;
+    }
+
+    /**
+     * @param Product $product
+     * @param int $categoryId
+     * @param array<int, int[]> $deletedCategoryIdsByParent
+     */
+    private function addProductDefaultCategory(Product $product, int $categoryId, array $deletedCategoryIdsByParent): void
+    {
+        // @todo: inject rootCategoryId into constructor
+        $rootCategoryId = (int) Configuration::get('PS_ROOT_CATEGORY');
+
+        $parentCategoryId = $this->findCategoryParentId($categoryId, $deletedCategoryIdsByParent);
+        $product->id_category_default = $parentCategoryId ?: $rootCategoryId;
+        $product->addToCategories($parentCategoryId);
+        $product->save();
+    }
+
+    /**
+     * @param array<int, int[]> $deletedCategoryIdsByParent
+     * @param CategoryDeleteMode $mode
+     */
+    private function updateProductsWithoutCategories(array $deletedCategoryIdsByParent, CategoryDeleteMode $mode): void
+    {
+        $productsWithoutCategories = $this->findProductsWithoutCategories();
+
+        foreach ($productsWithoutCategories as $productWithoutCategory) {
+            //@todo: use ProductRepository->get()?
+            $product = new Product((int) $productWithoutCategory['id_product']);
+
+            if (!$product->id) {
+                continue;
+            }
+
+            if ($mode->shouldRemoveProducts()) {
+                $product->delete();
+
+                continue;
+            }
+
+            if ($mode->shouldDisableProducts()) {
```

Comment:
So it means this mode was never handled correctly?

Reply:
Hmm I actually don't understand why would we want to delete product if parent category id = 0 :thinking: 
The only legit case when category doesn't have parent is when its root category (isnt it?). But why would we want to delete such product?

Reply:
the scenario you are talking about has a related deletion mode to it :arrow_up: (you can choose to remove the products).
But how it used to work before is - if you check this https://github.com/PrestaShop/PrestaShop/pull/30231/files#diff-6522f54552ae3b1e5495e7835ccf33da060205734c9dd07368b8af46f5a7e5f1R102
it would delete product if parent category id 0 is provided. So idk, just seems to be abit random :smile: